app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,33 @@
+class PostsController < ApplicationController
+
+  def create
+    @user = User.find(params[:user_id])
+    @user.posts.build(post_params)
+    if @user.save
+      flash.notice = "Post created."
+      redirect_back(fallback_location: current_user)
+    else
+      flash.notice = "Error. Post not created."
+      redirect_back(fallback_location: current_user)
+    end
+  end
+
+  def destroy
```

Comment:
by not using current user in this, I think you leave yourself open to someone being to change the params within the element and then delete posts that are not theirs. I'll check it out in the live app and let you know.